### DISCUSSION TITLE
How to keep built-in labels (crossplane-kind, crossplane-name, and crossplane-provider) when specifying user specified labels ?

### DISCUSSION THREAD

**Discussion by gberche-orange** [crossplane/crossplane#4296] [Q&A]
I'm trying to add user labels on my gcp cloudsql instance, in addition to the built-in labels crossplane-kind, crossplane-name, and crossplane-provider, (specified into https://github.com/crossplane/crossplane/blob/master/design/one-pager-managed-resource-api-design.md#external-resource-labeling )

However, as soon as new labels are defined in the MR spec, then the built-in labels get removed from the GCP cloud sql api, and only the explicit user labels get sent to the GCP sql api. I did not yet find associated integration test case into the upbound gcp-provider repo (only examples of user defined labels in https://github.com/upbound/provider-gcp/blob/bc2a0fa01b445d86af4234881490850df58b992a/examples-generated/monitoring/service.yaml#L16-L18, but no yet kuttl assertions about the built-in labels).

Any idea for a workaround ?

(initially posted on slack general channel https://crossplane.slack.com/archives/CEG3T90A1/p1688485153373599 )

**Comment by jbw976** [comment]
This sounds like a bug to me, I wouldn't expect user labels to overwrite the crossplane built-in labels that get applied automatically 🤔 

Not sure of a workaround here though, just expressing that this doesn't sound like correct or expected behavior to me at first glance 😅

**Reply by gberche-orange** [reply]
Thanks @jbw976 
Would you mind converting this discussion into an issue as to keep history?

I don't know whether the root cause is in the upbound gcp provider, upjet or possibly crossplane runtime.

**Reply by jbw976** [reply]
Transferred this to #4302 :)
